### PR TITLE
Migrate pkg/kubelet/token to structured logging

### PR DESCRIPTION
**Pull request by palnabarun**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Migrate pkg/kubelet/token to structured logging

#### Which issue(s) this PR fixes:

Part of #98976 

#### Special notes for your reviewer:

NONE

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging
```

/sig node instrumentation
/priority important-soon

### CODE REVIEW

File: pkg/kubelet/token/token_manager.go [hunk 1]
```diff
@@ -118,7 +118,7 @@ func (m *Manager) GetServiceAccountToken(namespace, name string, tr *authenticat
 		case m.expired(ctr):
 			return nil, fmt.Errorf("token %s expired and refresh failed: %v", key, err)
 		default:
-			klog.Errorf("couldn't update token %s: %v", key, err)
+			klog.ErrorS(err, "couldn't update token", "token", name)
```
Comment (adisky) [inline pkg/kubelet/token/token_manager.go:121]:
```suggestion
			klog.ErrorS(err, "Couldn't update token", "key",  key)
```

File: pkg/kubelet/token/token_manager.go [hunk 2]
```diff
@@ -172,7 +172,7 @@ func (m *Manager) requiresRefresh(tr *authenticationv1.TokenRequest) bool {
 	if tr.Spec.ExpirationSeconds == nil {
 		cpy := tr.DeepCopy()
 		cpy.Status.Token = ""
-		klog.Errorf("expiration seconds was nil for tr: %#v", cpy)
+		klog.ErrorS(errors.New("expiration seconds was nil"), "failed to check if token requires refresh", "tokenRequest", cpy)
```
Comment (adisky) [inline pkg/kubelet/token/token_manager.go:175]:
```suggestion
		klog.ErrorS(nil, "Expiration seconds was nil for token request", "tokenRequest", cpy)
```

**Comment by palnabarun** [comment]
@adisky -- Resolved both. Thanks for the review! :)

**Comment by adisky** [comment]
@palnabarun Thanks!!
/cc @serathius

File: pkg/kubelet/token/token_manager.go [hunk 3]
```diff
@@ -118,7 +118,7 @@ func (m *Manager) GetServiceAccountToken(namespace, name string, tr *authenticat
 		case m.expired(ctr):
 			return nil, fmt.Errorf("token %s expired and refresh failed: %v", key, err)
 		default:
-			klog.Errorf("couldn't update token %s: %v", key, err)
+			klog.ErrorS(err, "Couldn't update token", "key", key)
```
Comment (serathius) [inline pkg/kubelet/token/token_manager.go:121]:
```suggestion
			klog.ErrorS(err, "Couldn't update token", "cacheKey", key)
```

Comment (shubheksha) [reply pkg/kubelet/token/token_manager.go:121]: @palnabarun could you address this? It'll be ready to go after that :)

Comment (palnabarun) [reply pkg/kubelet/token/token_manager.go]: @shubheksha -- thanks for the ping! Apologies for the delay. The suggestion is applied now.

I had filed this one as a live demo for a Contributor Workshop and it went out of my context after that.

**Review by ehashman — COMMENTED** [review]
/lgtm

**Pull request merged**